packages/cdktf/lib/terraform-iterator.ts

```diff
@@ -284,3 +305,47 @@ export class MapTerraformIterator extends TerraformIterator {
     return this._getValue();
   }
 }
+
+// eslint-disable-next-line jsdoc/require-jsdoc
+export class ResourceTerraformIterator extends TerraformIterator {
+  constructor(private readonly element: ITerraformResource) {
+    super();
+
+    if (element.count) {
+      throw new Error(
+        "Cannot create iterator from resource with count argument. Please use the same TerraformCount used in the resource passed here instead."
+      );
+    }
+
+    if (!element.forEach) {
+      console.log({ element });
+      throw new Error(
+        "Cannot create iterator from resource without for_each argument"
+      );
+    }
+  }
+
+  /**
+   * Returns the currenty entry in the list or set that is being iterated over.
+   * For lists this is the same as `iterator.value`. If you need the index,
+   * use count using the escape hatch:
```

Comment:
> using the escape hatch

Don't we already have support for `TerraformCount` nowadays?

Reply:
Yeah, I'll fix the comment from where I copied it over from as well 👍